framework/test/unit/modules/token/api.spec.ts

```diff
@@ -0,0 +1,417 @@
+		});
+
+		it('should reject amount is less than zero', async () => {
+			await expect(
+				api.burn(apiContext, defaultAddress, defaultTokenID, BigInt(-1)),
+			).rejects.toThrow('Amount must be a positive integer to burn');
+		});
+
+		it('should reject if address does not have enough balance', async () => {
+			await expect(
+				api.burn(
+					apiContext,
+					defaultAddress,
+					defaultTokenID,
+					defaultAccount.availableBalance + BigInt(1),
+				),
+			).rejects.toThrow('is not sufficient for');
+		});
+
+		it('should update address balance and total supply', async () => {
+			await expect(
+				api.burn(apiContext, defaultAddress, defaultTokenID, defaultAccount.availableBalance),
+			).resolves.toBeUndefined();
+			await expect(
+				api.getAvailableBalance(apiContext, defaultAddress, defaultTokenID),
+			).resolves.toEqual(BigInt(0));
+
+			const supplyStore = apiContext.getStore(MODULE_ID_TOKEN, STORE_PREFIX_SUPPLY);
+			const { totalSupply } = await supplyStore.getWithSchema<SupplyStoreData>(
+				defaultTokenIDAlias.slice(CHAIN_ID_LENGTH),
+				supplyStoreSchema,
+			);
+			expect(totalSupply).toEqual(defaultTotalSupply - defaultAccount.availableBalance);
+		});
+	});
+
+	describe('lock', () => {
+		it('should reject amount is less than zero', async () => {
+			await expect(
+				api.lock(apiContext, defaultAddress, 12, defaultTokenID, BigInt(-1)),
+			).rejects.toThrow('Amount must be a positive integer to lock');
+		});
+
+		it('should reject if address does not have enough balance', async () => {
+			await expect(
+				api.lock(
+					apiContext,
+					defaultAddress,
+					12,
+					defaultTokenID,
+					defaultAccount.availableBalance + BigInt(1),
+				),
+			).rejects.toThrow('is not sufficient for');
+		});
+
+		it('should update address balance', async () => {
+			await expect(
+				api.lock(apiContext, defaultAddress, 2, defaultTokenID, defaultAccount.availableBalance),
+			).resolves.toBeUndefined();
+			await expect(
+				api.getAvailableBalance(apiContext, defaultAddress, defaultTokenID),
+			).resolves.toEqual(BigInt(0));
+		});
+
+		it('should update locked balances to be sorted by module ID', async () => {
+			await expect(
+				api.lock(apiContext, defaultAddress, 2, defaultTokenID, defaultAccount.availableBalance),
+			).resolves.toBeUndefined();
+			const userStore = apiContext.getStore(MODULE_ID_TOKEN, STORE_PREFIX_USER);
+			const { lockedBalances } = await userStore.getWithSchema<UserStoreData>(
+				getUserStoreKey(defaultAddress, defaultTokenIDAlias),
+				userStoreSchema,
+			);
+			expect(lockedBalances[0].moduleID).toEqual(2);
+		});
+	});
+
+	describe('unlock', () => {
+		it('should reject amount is less than zero', async () => {
+			await expect(
+				api.unlock(apiContext, defaultAddress, 12, defaultTokenID, BigInt(-1)),
+			).rejects.toThrow('Amount must be a positive integer to unlock');
+		});
+
+		it('should reject if address does not have corresponding locked balance', async () => {
```

Comment:
The description seems confusing to me, not able to follow what it is testing with the error message.